web/docs/language/features.md

```diff
@@ -178,138 +178,246 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
+- Wasp **generates a server-side NodeJS function** that shares its name with the Query. This function's interface is identical to the client-side function from the previous point.
+
+Generating two such functions ensures a uniform calling interface across the entire app (both backend and frontend).
 
-By declaring a NodeJS function as a Wasp query, following happens:
-- Wasp generates HTTP API route on the NodeJS server that calls the NodeJS query function.
-- Wasp generates JS function on the client that has the name under which query was declared and takes same arguments as the NodeJS query function. Internally it uses above mentioned HTTP API route to call the NodeJS query function.
 
-On client, you can import generated query JS function as `import getTasks from '@wasp/queries/getTasks.js'`.
-Then, you can either use it directly, or you can use it via special `useQuery` React hook (provided by Wasp**) to make it reactive.
+#### Using the Query
+To use the Query, you can import it from `@wasp` and call it directly. As mentioned, the usage is the same regardless of whether you're on the server or the client:
+```javascript
+import fetchAllTasks from '@wasp/queries/fetchAllTasks.js'
+import fetchFilteredTasks from '@wasp/queries/fetchFilteredTasks.js'
 
-On server, you can import it the same way as on client, and then you can call it directly.
+// ...
 
-**NOTE**: Wasp will not stop you from importing NodeJS function directly on server, e.g. `import { getAllTasks } from "./foo.js"`, but you shouldn't do it, because it will import pure NodeJS function and not a query recognized by Wasp, so it will not get all the features of a Wasp query.
+const allTasks = await fetchAllTasks();
+const doneTasks = await fetchFilteredTasks({isDone: true})
+```
 
-#### useQuery
-`useQuery` hook provided by Wasp is actually just a thin wrapper for `useQuery` hook from [react-query](https://github.com/tannerlinsley/react-query).
+**NOTE**: Wasp will not stop you from importing a Query's NodeJS implementation from `./queries.js` and calling it directly. However, we advise against this, as you'll lose all the useful features a Wasp Query provides (e.g., entity injection).
 
-You can import it as `import { useQuery } from '@wasp/queries'`.
+#### The `useQuery` hook
+When using Queries on the client, you can make them reactive with the help of the `useQuery` hook.
+This hook comes bundled with Wasp and is a thin wrapper around the `useQuery` hook from [_react-query_](https://github.com/tannerlinsley/react-query).
 
-Wasp `useQuery` takes three args:
-- `queryFn`: client query function generated by Wasp based on query declaration, e.g. one you get by importing in JS like this: `import getTasks from '@wasp/queries/getTasks.js'`.
-- `queryFnArgs`
-- `config`: react-query `config`.
+Wasp's `useQuery` hook takes three arguments:
```

Comment:
```suggestion
Wasp's `useQuery` hook accepts three arguments:
```
To make it sound more fancy :D But no need really to change it, I'm just writing things as they come to my mind :D